day2OfJS/day2.js

```diff
@@ -0,0 +1,262 @@
+let phrase = 'I hope this course is not full of jargon';
+console.log(phrase.match('jargon'));
+*/
+
+/*
+7. Generate a random number between 0 and 100 inclusively.
+const num = Math.floor(Math.random() * 101);
+console.log(num);
+*/
+
+/*
+8. Generate a random number between 50 and 100 inclusively.
+const num = Math.floor(Math.random() * 51) + 50;
+console.log(num);
+*/
+
+/*
+9. Generate a random number between 0 and 255 inclusively.
+const num = Math.floor(Math.random() * 256);
+console.log(num);
+*/
+
+/*
+10. Access the 'JavaScript' string characters using a random number.
+let randomString = 'JavaScript';
+let random = Math.floor(Math.random() * randomString.length);
+console.log(random);
+*/
+
+/*
+11. Use console.log() and escape characters to print the following pattern.
+1 1 1 1 1
+2 1 2 4 8
+3 1 3 9 27
+4 1 4 16 64
+5 1 5 25 125
+console.log('1\t1\t1\t1\t1\n2\t1\t2\t4\t8\n3\t1\t3\t9\t27\n4\t1\t4\t16\t64\n5\t1\t5\t25\t125');
+*/
+
+/*
+12. Use substr to slice out the phrase 'You cannot end a sentence with because because because is a conjunction' from the following sentence:'You cannot end a sentence with because because because is a conjunction'
+let phrase = 'You cannot end a sentence with because because because is a conjunction';
+console.log(phrase.substr(31, 24));
+*/
+
+
+/*
+Exercises: Level 3
+1. 'Love is the best thing in this world. Some found their love and some are still looking for their love.' Count the number of word love in this sentence.
+let love = 'Love is the best thing in this world. Some found their love and some are still looking for their love.';
+console.log(love.match(/love/gi));
+*/
+
+
+/*
+2. Use match() to count the number of all because in the following sentence:'You cannot end a sentence with because because because is a conjunction'
+let because = 'You cannot end a sentence with because because because is a conjunction';
+console.log(because.match(/because/gi));
+*/
+
+
+/*
+3. Clean the following text and find the most frequent word(hint, use replace and regular expressions).
+const sentence = '%I $am@% a %tea@cher%, &and& I lo%#ve %te@a@ching%;. The@re $is no@th@ing; &as& mo@re rewarding as educa@ting &and& @emp%o@weri@ng peo@ple. ;I found tea@ching m%o@re interesting tha@n any ot#her %jo@bs./%Do@es thi%s mo@tiv#ate yo@u to be a tea@cher!? %Th#is 30#Days&OfJavaScript &is al@so $the $resu@lt of &love& of tea&ching';
+console.log(sentence.replace(/[^a-zA-Z 0-9.]+/gi, ''));
+*/
+
+
+/* 
+4. Calculate the person's total annual income by extracting the figures from the following text. 
+'He earns 5000 euros of salary per month, 10000 euros of annual bonus, 15000 euros of online courses per month'.
+let sum = 'He earns 5000 euros of salary per month, 10000 euros of annual bonus, 15000 euros of online courses per month';
+let matches = sum.match(/(\d+)/g)
+let mult1 = Number(matches[0]) * 12
+let mult2 = Number(matches[1])
+let mult3 = Number(matches[2]) * 12
```

Comment:
The name of the variables need to have a name that helps to understand whats happening in the code, so:
eg.
`sum`: I expect a sum but I get a text;
`matches`: is too generic
`mult#`:  It do not have much context about the excercise